cs-commons/src/main/java/io/cloudslang/content/utils/NumberUtilities.java

```diff
@@ -0,0 +1,132 @@
+        }
+        if (!isValidInt(integerStr)) {
+            return false;
+        }
+        final int aInteger = toInteger(integerStr);
+        return lowerBound < aInteger && aInteger < upperBound;
+    }
+
+    /** Given an integer string if it's a valid int (see isValidInt) it converts it into an integer otherwise it throws an exception
+     * @param integerStr the integer to convert
+     * @return the integer value of the integerStr
+     * @throws IllegalArgumentException if the passed integer string is not a valid integer
+     */
+    public static int toInteger(@Nullable final String integerStr) {
+        if (!isValidInt(integerStr)) {
+            throw new IllegalArgumentException(integerStr + ExceptionsValues.EXCEPTION_DELIMITER + ExceptionsValues.INVALID_INTEGER_VALUE);
+        }
+        final String stripedInteger = StringUtils.strip(integerStr);
+        return NumberUtils.createInteger(stripedInteger);
+    }
+
+    /** If the integer string is null or empty, it returns the defaultInteger otherwise it returns the integer value (see toInteger)
+     * @param integerStr the integer to convert
+     * @param defaultInteger the default value if the integerStr is null or the empty string
+     * @return the integer value of the string or the defaultInteger is the integer string is empty
+     * @throws IllegalArgumentException if the passed integer string is not a valid integer
+     */
+    public static int toInteger(@Nullable final String integerStr, final int defaultInteger) {
+        return StringUtils.isNoneEmpty(integerStr) ? toInteger(integerStr) : defaultInteger;
+    }
+
+    /** Given an double string, it checks if it's a valid double (based on apaches NumberUtils.createDouble)
+     * @param doubleStr the double string to check
+     * @return true if it's valid, otherwise false
+     */
+    public static boolean isValidDouble(@Nullable final String doubleStr) {
+        if (StringUtils.isBlank(doubleStr)){
+            return false;
+        }
+        final String stripedDouble = StringUtils.strip(doubleStr);
+        try {
+            NumberUtils.createDouble(stripedDouble);
+            return true;
+        } catch (NumberFormatException e) {
+            return false;
+        }
+    }
+
+    /** Given an double string, it checks if it's a valid double (base on apaches NumberUtils.createDouble) and if
+     * it's between the lowerBound and upperBound (without including them)
+     * @param doubleStr the integer string to check
+     * @param lowerBound the lower bound of the interval
+     * @param upperBound the upper bound of the interval
+     * @return true if the integer string is valid and in between the lowerBound and upperBound, false otherwise
+     * @throws IllegalArgumentException if the lowerBound is not less than the upperBound
+     */
+    public static boolean isValidDouble(@Nullable final String doubleStr, double lowerBound, double upperBound) {
+        if (lowerBound >= upperBound) {
```

Comment:
this is not correct, use Double.compare or something from apache for double comparison. Comparing with operators is very error prone for floating point arithmetic